api/client/volume/client.go

```diff
@@ -708,3 +708,15 @@ func (v *volumeClient) CloudMigrateStatus() (*api.CloudMigrateStatusResponse, er
 	}
 	return statusResponse, nil
 }
+
+// Du specified volume id and specifically path (if provided)
+func (v *volumeClient) Catalog(id, subfolder, maxDepth string) (api.CatalogResponse, error) {
+	var catalog api.CatalogResponse
+
+	req := v.c.Get().Resource(volumePath + "/catalog").Instance(id)
+	if err := req.QueryOption("subfolder", subfolder).QueryOption("maxdepth", maxDepth).Do().Unmarshal(&catalog); err != nil {
```

Comment:
nit: use the OptSubFolder constants. Also in the client you are using "maxdepth" but in the server the constant for "depth" is being used.